### PR TITLE
make imgcodecs optional for highgui

### PR DESCRIPTION
**Pull request by sturkmen72**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/highgui/src/plugin_wrapper.impl.hpp [hunk 1]
```diff
@@ -235,12 +235,9 @@ std::vector<FileSystemPath_t> getPluginCandidates(const std::string& baseName)
     return results;
 }
 
-// NB: require loading of imgcodecs module
-static void* g_imwrite = (void*)imwrite;
 
 void PluginUIBackendFactory::loadPlugin()
 {
-    CV_Assert(g_imwrite);
```
Comment (alalek) [inline modules/highgui/src/plugin_wrapper.impl.hpp:243]: Please keep under `#ifdef` instead.

BTW, Removal/disabling of this code requires `OPENCV_PLUGIN_NO_LINK=OFF` to force plugin linking to opencv binaries.

File: modules/highgui/src/window_w32.cpp [hunk 2]
```diff
@@ -2193,6 +2194,10 @@ static void showSaveDialog(CvWindow& window)
         cv::flip(cv::Mat(sz.cy, sz.cx, CV_8UC(channels), data, (sz.cx * channels + 3) & -4), tmp, 0);
         cv::imwrite(szFileName, tmp);
     }
+#else
+    CV_UNUSED(window);
+    return;
```
Comment (alalek) [inline modules/highgui/src/window_w32.cpp:2199]: It makes sense to emit some `CV_LOG_WARNING()` for this case.

**Comment by asmorkalov** [comment]
@alalek Could we merge this?

**Review by vpisarev — APPROVED** [review]

**Comment by vpisarev** [comment]
@asmorkalov, I think so

**Review by alalek — APPROVED** [review]

**Pull request merged**